lang/test/org/partiql/lang/syntax/SqlParserTestBase.kt

```diff
@@ -45,82 +41,79 @@ abstract class SqlParserTestBase : TestBase() {
     protected fun parse(source: String): ExprNode = parser.parseExprNode(source)
     protected fun parseToAst(source: String): PartiqlAst.Statement = parser.parseAstStatement(source)
 
+    /**
+     * This method is used by test cases, to test with PIG AST, while the expected PIG AST is a PIG builder
+     */
     protected fun assertExpression(
         source: String,
-        pigBuilder: PartiqlAst.Builder.() -> PartiqlAst.PartiqlAstNode
+        expectedPigBuilder: PartiqlAst.Builder.() -> PartiqlAst.PartiqlAstNode
     ) {
-        val parsedExprNode = parse(source)
-
-        val expectedPartiQlAst = PartiqlAst.build { pigBuilder() }.toIonElement().toString()
-        // Convert the query to ExprNode
+        val expectedPigAst = PartiqlAst.build { expectedPigBuilder() }.toIonElement().toString()
 
-        val partiqlAst = loadIonSexp(expectedPartiQlAst)
-        partiqlAssert(parsedExprNode, partiqlAst, source)
-
-        pigDomainAssert(parsedExprNode, partiqlAst.toIonElement().asSexp())
-        pigExprNodeTransformAsserts(parsedExprNode)
+        assertExpression(source, expectedPigAst)
     }
 
-    // TODO: refactor the signature with pig builder
+    /**
+     * This method is used by test cases, to test with PIG AST, while the expected PIG AST is a string
+     */
     protected fun assertExpression(
```

Comment:
Since this PR is about the changes on which are not relevant to `ExprNode` deprecation, it is assumed that `ExprNode` has not been deprecated yet. Later when we deprecate it, I will remove the `ExprNode` here.

Reply:
Can you update the `kdoc` for this `assertExpression` to reflect what it does? For example, it also checks that certain round trips are valid.

Reply:
Done in the next commit with comment 'Added more kdocs'.